src/components/Tile.tsx

```diff
@@ -0,0 +1,52 @@
+import { Link, useFirebaseAnalytics } from '@ugrc/utah-design-system';
+import { Check } from 'lucide-react';
+import { useEffect, useRef } from 'react';
+import { twJoin } from 'tailwind-merge';
+import useTiles from '../hooks/useTiles';
+import type { TileFeature } from '../types';
+
+export type TileProps = {
+  attributes: TileFeature['attributes'];
+  onHover: (objectId: number, on: boolean) => void;
+  isHighlighted?: boolean;
+};
+
+export default function Tile({ attributes, onHover, isHighlighted }: TileProps) {
+  const { OBJECTID, TILE, SIZE, EXT, PATH } = attributes;
+  const filename = `${TILE}${EXT}`;
+
+  const rootNodeRef = useRef<HTMLDivElement>(null);
+  const logEvent = useFirebaseAnalytics();
+  const { markAsDownloaded, downloadedTiles } = useTiles();
+
+  useEffect(() => {
+    if (isHighlighted) {
+      rootNodeRef.current?.scrollIntoView({ block: 'nearest', behavior: 'smooth' });
+    }
+  }, [isHighlighted]);
+
+  return (
+    <div ref={rootNodeRef}>
+      <Link
+        download
+        className={twJoin(
+          'flex items-center justify-between rounded p-1 no-underline',
+          isHighlighted && 'bg-zinc-50 font-bold dark:bg-zinc-800',
+        )}
+        href={`${PATH}${filename}`}
+        onClick={() => {
```

Comment:
I believe our design system and RAC uses onPress instead of onClick. I recommend changing all the instances of onClick to onPress since it handles press interactions across mouse, touch, keyboard, and screen readers. It normalizes behavior across browsers and platforms, and handles many nuances of dealing with pointer and keyboard events.

Reply:
I need to try and rewrite my brain to use onPress. Thanks for the reminder.